test/integration/puppetlabs/trapperkeeper/services/scheduler/scheduler_service_test.clj

```diff
@@ -62,37 +73,185 @@
     (with-app-with-empty-config app [scheduler-service]
       (let [service (tk/get-service app :SchedulerService)
             started (promise)
             stopped (promise)
-            start-time (atom 0)
-            completed (promise)
             job (fn []
-                  (reset! start-time (System/currentTimeMillis))
-                  (deliver started nil)
-                  (deref stopped)
-                  (deliver completed nil))
-            job-id (interspaced service 10 job)]
-        ; wait for the job to start
+                  (deliver started nil))
+            interval 10
+            group-id-0 :unique-group-id
+            group-id-1 :more-unique-group-id
+            ; create one job without a service id and two with one group-id
+            ; and a third with a different service id
+            job-3 (interspaced service interval (constantly true))
+            job-2 (interspaced service interval (constantly true) group-id-0)
+            job-1 (interspaced service interval (constantly true) group-id-0)
+            job-0 (interspaced service interval job group-id-1)]
+
+        (testing "all the jobs were started"
+          (is (= 4 (count-jobs service)))
+          (is (= 2 (count-jobs service group-id-0)))
+          (is (= 1 (count-jobs service group-id-1))))
+
+        ; wait for the jobs to start
         (deref started)
-        (let [original-start-time @start-time]
-          (testing "the job can be stopped"
-            (is (stop-job service job-id)))
-          (deliver stopped nil)
-          (deref completed)
-          ; wait a bit, ensure the job does not run again
-          (testing "the job should not run again"
-            (Thread/sleep 100)
-            (is (= original-start-time @start-time))))))))
+        (Thread/sleep 100)
+        (testing "stopping one group-id does not stop them all"
+          (stop-grouped-jobs service group-id-0)
+          (Thread/sleep 100)
+          (is (= 2 (count-jobs service)))
+          (is (= 0 (count-jobs service group-id-0)))
+          (is (= 1 (count-jobs service group-id-1))))
+
+        (testing "stopping one job does not stop the service id based job"
+          (stop-job service job-3)
+          (Thread/sleep 100)
+          (is (= 1 (count-jobs service)))
+          (is (= 0 (count-jobs service group-id-0)))
+          (is (= 1 (count-jobs service group-id-1))))
+
+        (testing "stopping by service id stops the job"
+          (stop-grouped-jobs service group-id-1))
+          (Thread/sleep 100)
+          (is (= 0 (count-jobs service)))
+          (is (= 0 (count-jobs service group-id-0)))
+          (is (= 0 (count-jobs service group-id-1)))))))
 
 (defn schedule-random-jobs
   "Schedules several random jobs and returns their IDs."
   [service]
   (set
     (for [x [1 2 3]]
-      (interspaced service 1000 (constantly x)))))
+      (:id (interspaced service 1000 (constantly x))))))
```

Comment:
Okay, that makes sense.  It seems a little strange that the at-at job is referred to as `:id` but I realize that's consistent with how things were done previously, so 👍 

This is an API change, so if this project was post-1.0, it would be a major version bump.  Given that it's still pre-1.0, perhaps it would just be best to bump the project version to `0.2.0-SNAPSHOT` as a part of this PR, so that the next release is at least a Y.